Makefile

```diff
@@ -0,0 +1,47 @@
+.PHONY: all install nopyc clean test docs local
```

Comment:
Quality of life improvements for you and anyone else trying to develop on this. Simply run `make test` and your dev env will be provisioned and tests run. `make local` will install the built version of the library in to your local `pip` cache. `make docs`, well, docs :).